examples/sw.impl.h

```diff
@@ -0,0 +1,362 @@
+  }
+  std::chrono::time_point<std::chrono::high_resolution_clock> beg, end;
+  std::cout << "Computing using serial version....\n";
+  beg = std::chrono::high_resolution_clock::now();
+  int val1 = SW_serial(a, b);
+  end = std::chrono::high_resolution_clock::now();
+  std::cout << "....done!" << std::endl;
+  std::cout << "Serial Execution Time (milliseconds) : "
+            << (std::chrono::duration_cast<std::chrono::microseconds>(end - beg).count()) / 1e3 << std::endl;
+
+  //int val2 = SW_OpenMP(a, b, r, base_size);
+  ttg_initialize(argc, argv, -1);
+
+  Edge<Key, BlockMatrix<int>> leftedge, topedge, diagedge;
+  Edge<Key, int> resultedge;
+  auto s = make_sw1(sw_iterative<int>, block_size, a, b, problem_size, leftedge, topedge,
+                  diagedge, resultedge);
+  auto s1 = make_sw2(sw_iterative<int>, block_size, a, b, problem_size, leftedge, topedge,
+                  diagedge, resultedge);
+  auto r = make_result(verify, val1, resultedge);
+
+  auto connected = make_graph_executable(s.get());
+  assert(connected);
+  TTGUNUSED(connected);
+  std::cout << "Graph is connected.\n";
+
+  if (ttg_default_execution_context().rank() == 0) {
+    //std::cout << "==== begin dot ====\n";
+    //std::cout << Dot()(s.get()) << std::endl;
+    //std::cout << "==== end dot ====\n";
+
+    beg = std::chrono::high_resolution_clock::now();
+    s->in<0>()->send(Key(0,0), BlockMatrix<int>());
+  }
+
+  ttg_execute(ttg_default_execution_context());
+  ttg_fence(ttg_default_execution_context());
+  end = std::chrono::high_resolution_clock::now();
+  std::cout << "TTG Execution Time (milliseconds) : "
+              << (std::chrono::duration_cast<std::chrono::microseconds>(end - beg).count()) / 1000 << std::endl;
+
+}
+
+int SW_serial(const std::string &a, const std::string &b) {
+   size_t n = a.length();
+   int *X = new int[n*n]; // (int*) malloc(n*n*sizeof(int));
+   for(size_t i = 0; i < n; ++i) { // updating the row X[i][...]
+      for(size_t j = 0; j < n; ++j) { // updating the cell X[i][j]
+         int left_value = (j > 0 ? X[i*n + j-1] : (i+1)*get_score(a[i],'_')) + 
+                            get_score('_', b[j]);
+
+         int top_value = (i > 0 ? X[(i-1)*n + j] : (j+1)*get_score('_', b[j])) + 
+                           get_score(a[i], '_');
+
+         int diag_value = get_score(a[i], b[j]);
+         if (i > 0 && j > 0) { diag_value += X[(i-1)*n + j-1]; }
+         else if (i > 0) { diag_value += (i)*get_score(a[i],'_');  }
+         else if (j > 0) { diag_value += (j)*get_score('_', b[j]); }
+
+         X[i*n+j] = std::max({left_value, top_value, diag_value});
+        //std::cout << left_value << " " << top_value << " " << diag_value << "-->" << X[i*n+j] << " ";
+      }
+      //std::cout << std::endl;
+   }
+   
+   // returning the data at the bottom-right as the final value
+   int final_value = X[(n-1)*n+(n-1)];
+   delete[] X;
+   return final_value;
+}
+
+/*int SW_OpenMP(const std::string &a, const std::string &b,
```

Comment:
Here a few commented out functions

Reply:
This code is the OpenMP implementation on Smith Waterman. This is kept for future use.